### PR TITLE
Update console to use core.http instead of jQuery.ajax

### LINKED ISSUE #3044: change console plugin to use core.http to replace jQuery.ajax
**Is your feature request related to a problem? Please describe.**
OSD DevTool console browser side plugin uses `jQuery.ajax` to call NodeJS API instead of using `core.http` which is inconsistent with all other use cases, all settings on browser side `core.http`, like request interceptors don't apply to the devtool console calls. related code: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/console/public/lib/opensearch/opensearch.ts#L61

**Describe the solution you'd like**
refactor console plugin to use `core.http` to call OSD server, get rid of 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by zengyan-amazon**
Signed-off-by: Yan Zeng <zengyan@amazon.com>

### Description
Update console plugin browser side to use `core.http` when calling OSD server APIs, get rid of use of jQuery.ajax.
 
### Issues Resolved
#3044 
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts [hunk 1]
```diff
@@ -94,79 +96,94 @@ export function sendRequestToOpenSearch(
...
+          args.http,
+          opensearchMethod,
+          opensearchPath,
+          opensearchData
+        );
+        if (reqId !== CURRENT_REQ_ID) {
+          return;
+        }
+        const statusCode = httpResponse.response?.status;
+        const isSuccess = statusCode && (statusCode >= 200 || statusCode === 404);
```
Comment (joshuarrrr) [inline src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts:110]: Doesn't `statusCode >= 200` imply that we'll consider all `3xx, 4xx, 5xx` status codes to be successful, too? The previous implementation was `>= 200 && <300`

**Review by joshuarrrr — COMMENTED** [review]
One question about status code checking. I see there aren't any new tests here. Are our existing tests complete and robust enough to validate the safety of this change?

File: src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts (hunk 1 above)
Comment (zengyan-amazon) [reply src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts:110]: oh, it should be `>= 200 && <300`, will fix it and add a few more test cases in next revision

File: src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts [hunk 2]
```diff
@@ -94,79 +96,95 @@ export function sendRequestToOpenSearch(
...
+          opensearchMethod,
+          opensearchPath,
+          opensearchData
+        );
+        if (reqId !== CURRENT_REQ_ID) {
+          return;
+        }
+        const statusCode = httpResponse.response?.status;
+        const isSuccess =
+          statusCode && ((statusCode >= 200 && statusCode < 300) || statusCode === 404);
```
Comment (kristenTian) [inline src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts:111]: minor, should we keep the comments for `404`? // Things like DELETE index where the index is not there are OK.

And do we need check if  httpResponse is undefined -> `httpResponse?.response?.status`

**Review by joshuarrrr — APPROVED** [review]

File: src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts [hunk 3]
```diff
@@ -94,79 +96,95 @@ export function sendRequestToOpenSearch(
...
+        if (isSuccess) {
+          const contentType = httpResponse.response.headers.get('Content-Type') as BaseResponseType;
+          let value = '';
+          if (contentType.includes('application/json')) {
+            value = JSON.stringify(httpResponse.body, null, 2);
```
Comment (kristenTian) [inline src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts:117]: `JSON.stringify(httpResponse.body, null, 2)` => 2 spaces? is this consistent with current behavior? would this be the layout rendering on the dev console?

Comment (zengyan-amazon) [reply src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts:117]: yes, indentation of 2 spaces is expected

**Review by kristenTian — APPROVED** [review]

**Pull request merged**